xarray/conventions/netcdftimeindex.py

```diff
@@ -106,9 +120,23 @@ def get_date_type(self):
     return type(self._data[0])
 
 
+def assert_all_same_netcdftime_datetimes(data):
+    from netcdftime._netcdftime import datetime
+
+    if not isinstance(data[0], datetime):
+        raise TypeError(
+            'NetCDFTimeIndex requires netcdftime._netcdftime.datetime'
+            ' objects.')
```

Comment:
nit: I usually prefer to leave spaces at the lines instead of the the start of lines -- I think it looks slightly nicer.